core/state/sync_test.go

```diff
@@ -628,13 +622,6 @@ func TestIncompleteStateSync(t *testing.T) {
 		}
 		batch.Write()
 
-		for _, root := range nodehashes {
```

Comment:
It's a very hacky check so that I remove it.

What it does is: iterate the newly added nodes, treat them as the root node of the "sub-trie" and traverse the "sub-trie" respectively. This notion is not compatible in path-based also.